mobilecoind-dev-faucet/src/bin/main.rs

```diff
@@ -0,0 +1,317 @@
+#![deny(missing_docs)]
+
+//! HTTP faucet service backed by mobilecoind
+
+#![feature(proc_macro_hygiene, decl_macro)]
+
+use clap::Parser;
+use grpcio::ChannelBuilder;
+use mc_account_keys::AccountKey;
+use mc_api::printable::PrintableWrapper;
+use mc_common::logger::{create_app_logger, log, o, Logger};
+use mc_mobilecoind_api::{mobilecoind_api_grpc::MobilecoindApiClient, MobilecoindUri};
+use mc_mobilecoind_dev_faucet::{data_types::*, worker::Worker};
+use mc_transaction_core::{ring_signature::KeyImage, TokenId};
+use mc_util_grpc::ConnectionUriGrpcioChannel;
+use mc_util_keyfile::read_keyfile;
+use mc_util_serial::JsonU64;
+use rocket::{get, post, routes, serde::json::Json};
+use std::{collections::HashMap, path::PathBuf, sync::Arc, time::Duration};
+
+/// Command line config, set with defaults that will work with
+/// a standard mobilecoind instance
+#[derive(Clone, Debug, Parser)]
+#[clap(
+    name = "mobilecoind-dev-faucet",
+    about = "A stateless HTTP faucet server, backed by mobilecoind"
+)]
+pub struct Config {
+    /// Path to json-formatted key file, containing mnemonic or root entropy.
+    #[clap(long, env = "MC_KEYFILE")]
+    pub keyfile: PathBuf,
+
+    /// The amount factor, which determines the size of the payment we make. The
+    /// minimum fee is multiplied by this.
+    #[clap(long, default_value = "20", env = "MC_AMOUNT_FACTOR")]
+    pub amount_factor: u64,
+
+    /// Host to listen on.
+    #[clap(long, default_value = "127.0.0.1", env = "MC_LISTEN_HOST")]
+    pub listen_host: String,
+
+    /// Port to start webserver on.
+    #[clap(long, default_value = "9090", env = "MC_LISTEN_PORT")]
+    pub listen_port: u16,
+
+    /// MobileCoinD URI.
+    #[clap(
+        long,
+        default_value = "insecure-mobilecoind://127.0.0.1/",
+        env = "MC_MOBILECOIND_URI"
+    )]
+    pub mobilecoind_uri: MobilecoindUri,
+
+    /// Target Queue Depth. When the queue for a token id is less than this in
+    /// depth, the worker attempts to make a split Tx to produce more TxOuts
+    /// for the queue.
+    #[clap(long, default_value = "20", env = "MC_TARGET_QUEUE_DEPTH")]
+    pub target_queue_depth: usize,
+
+    /// Worker poll period in milliseconds.
+    #[clap(long, default_value = "100", env = "MC_WORKER_POLL_PERIOD_MS")]
+    pub worker_poll_period_ms: u64,
+}
+
+/// Connection to the mobilecoind client
+struct State {
+    /// The connection to mobilecoind
+    pub mobilecoind_api_client: MobilecoindApiClient,
+    /// The account key holding our funds
+    pub account_key: AccountKey,
+    /// The bytes of our monitor id, which holds the faucet's funds
+    pub monitor_id: Vec<u8>,
+    /// The public address of the faucet, which someone can use to replenish the
+    /// faucet
+    pub monitor_b58_address: String,
+    /// The amounts the faucet attempts to pay for each token id
+    /// This is initialized to network fee * amount factor at startup
+    pub faucet_amounts: HashMap<TokenId, u64>,
```

Comment:
Maybe more clarification these are the payout amounts?

```suggestion
    pub faucet_payout_amounts: HashMap<TokenId, u64>,
```

OR 

```suggestion
    pub faucet_payouts: HashMap<TokenId, u64>,
```